packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -570,6 +572,7 @@ class HotRunnerFactory {
     bool stayResident = true,
     bool ipv6 = false,
     FlutterProject? flutterProject,
+    Analytics? analytics,
```

Comment:
can we make this required? we shouldn't be adding implicit calls to globals.

Reply:
Yep, done!